### PR TITLE
Multiple Pod IPs

### PR DESCRIPTION
**Pull request by sarahalsmiller**
### Changes proposed in this PR:
- Assign multiple ips to gateways when there are multiple pods

### How I've tested this PR:
- Unit tests passing
- attempting to replicate circumstances on a kind cluster

### How I expect reviewers to test this PR:
- Unit tests passing 

### Checklist:
- [ X ] Tests added
- [ ] CHANGELOG entry added 
  > Run `make changelog-entry` for guidance in authoring a changelog entry, and
  > commit the resulting file, which should have a name matching your PR number.
  > Entries should use imperative present tense (e.g. Add support for...)

### CODE REVIEW

File: internal/k8s/reconciler/gateway.go [hunk 1]
```diff
@@ -221,22 +221,26 @@ func (g *K8sGateway) assignGatewayIPFromService(ctx context.Context, service *co
...
 		return nil
 	}
 
-	if pod.Status.PodIP != "" {
-		g.serviceReady = true
-		g.addresses = append(g.addresses, pod.Status.PodIP)
+	for _, pod := range pods {
+		if pod.Status.PodIP != "" {
+			g.serviceReady = true
+			if !ipInList(g.addresses, pod.Status.PodIP) {
```
Comment (nathancoleman) [inline internal/k8s/reconciler/gateway.go:240]: Is the check for duplicate IPs in list necessary? I think `g.addresses` is built up from scratch here every time, and the k8s API shouldn't return the same pod twice

Comment (sarahalsmiller) [reply internal/k8s/reconciler/gateway.go:240]: I can remove it, I mostly just added it because I already was adding it for the HostIP section

**Review by mikemorris — APPROVED** [review]
LGTM (just some additional minor `pod` -> `pods` renaming) - it appears the existing `internal/k8s/reconciler/gateway_test.go` tests are fully exercising that the expected host or pod IPs are being set on the gateway.

Probably worth adding a changelog entry?

File: .changelog/201.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:note
+Gateway ip address assignment logic updated to include the case when multiple different pod ips exist
```
Comment (nathancoleman) [inline .changelog/201.txt:2]:
```suggestion
Gateway IP address assignment logic updated to include the case when multiple different pod IPs exist
```

**Review by nathancoleman — APPROVED** [review]

**Pull request merged**